src/main/webapp/js/query.js

```diff
@@ -0,0 +1,328 @@
+    let profile = googleUser.getBasicProfile();
+    fetch(`/login?id_token=${id_token}`)
+        .then((response) => response.json())
+        .then((data) => {
+            localStorage.setItem("user", data.id);
+            localStorage.setItem("loggedIn", true);
+            addUserContent(profile.getName(), profile.getImageUrl());
+            toggleAccountMenu();
+        });
+}
+
+function addUserContent(name, image) {
+    document.getElementById("user-name").innerText = name;
+    document.getElementById("profile-pic").src = image;
+}
+
+function toggleAccountMenu() {
+    document.getElementById("account-menu").classList.toggle("show");
+    document.getElementById("sign-in").classList.toggle("hide");
+}
+
+function signOut() {
+    var auth2 = gapi.auth2.getAuthInstance();
+    auth2.signOut().then(function() {
+        console.log("User signed out.");
+    });
+    localStorage.setItem("user", 0);
+    toggleAccountMenu();
+}
+
+function toggleShow() {
+    document.getElementById("myDropdown").classList.toggle("show");
+}
+
+window.onclick = function(event) {
+    if (!event.target.matches(".dropbtn")) {
+        let dropdownEl = document.getElementById("myDropdown");
+        if (dropdownEl.classList.contains("show")) {
+            dropdownEl.classList.remove("show");
+        }
+    }
+};
+
+
+/*
+    SAVING SEARCHES
+*/
+function saveSearch(lat, lng, radius, keyword) {
+    let userID = 0;
+    if (localStorage.getItem("loggedIn")) {
+        userID = localStorage.getItem("user");
+    }
+    fetch(`/searches?user=${userID}&radius=${radius}&keywords=${keyword}&lat=${lat}&lng=${lng}`, {
+        method: "POST",
+    });
+}
+
+function getSearches() {
+    let userID = 0;
+    if (localStorage.getItem("loggedIn")) {
+        userID = localStorage.getItem("user");
+    }
+    fetch(`/searches?user=${userID}`, { method: "GET" })
+        .then((response) => response.json())
+        .then((searches) => {
+            const searchesEl = document.getElementById("cards");
+            searches.forEach((search) => {
+                searchesEl.appendChild(createSearchElement(search));
+            });
+        });
+}
+
+
+/*
+    HTML
+ */
+
+// AJAX POST for form
+$("#randomize-form").submit(function(event) {
+    const errorEl = document.getElementById("error");
+    errorEl.classList.add("hidden");
+
+    event.preventDefault();
+    let url = $(this).attr("action");
+    let lat = localStorage.getItem("lat");
+    let lng = localStorage.getItem("lng");
+    let queryStr = $(this).serialize() + `&lat=${lat}&lng=${lng}`;
+
+    $.ajax({
+        type: "POST",
+        url: url,
+        data: queryStr,
+        success: function(response) {
+            query();
+        }
+    });
+});
+
+// Form underline element
+$("input, textarea").blur(function() {
+    if ($(this).val() != "") {
+        $(this).addClass("active");
+    } else {
+        $(this).removeClass("active");
+    }
+});
+
+// TODO: make this seamless and non-jank
```

Comment:
FYI : not good practice to say "non-jank" TODOs should be descriptive enough so that the reader knows what to do

This is a good example: `TODO: remove this entirely; fallback should be to leave form blank instead`

Reply:
TODO changed